src/libs/elektra/keyset.c

```diff
@@ -1193,6 +1251,8 @@ Key * ksPop (KeySet * ks)
 
 	if (ks->size == 0) return 0;
 
+	elektraOpmphmInvalidate (ks);
```

Comment:
Isn't ksResize (additionally?) the better place for invalidation?

elektraKsPopAtCursor and KDB_O_POP should be invalidated, too.

Reply:
`ksResize (...)` is in my opinion a bad choice because it is deep inside the code and therefore hard to understand where it is invoked.
`elektraKsPopAtCursor (...)` uses `ksPop (...)` at the end.
The `KDB_O_POP` checks all invoke `elektraKsPopAtCursor (...)`.

Reply:
I wrote a test case for `elektraKsPopAtCursor (...)` if somebody changes it.

Reply:
Another argument against ksResize is that the method does not change any keys, so it may result in too much invalidation.